include/pisa/cursor/block_max_scored_cursor.hpp

```diff
@@ -1,24 +1,40 @@
 #pragma once
 
+#include <vector>
+
 #include "query/queries.hpp"
+#include "scored_cursor.hpp"
 #include "scorer/index_scorer.hpp"
 #include "wand_data.hpp"
-#include <vector>
 
 namespace pisa {
 
-template <typename Index, typename WandType>
-struct block_max_scored_cursor {
-    using enum_type = typename Index::document_enumerator;
-    using wdata_enum = typename WandType::wand_data_enumerator;
-
-    block_max_scored_cursor() = delete;
-
-    enum_type docs_enum;
-    wdata_enum w;
-    float q_weight;
-    term_scorer_t scorer;
-    float max_weight;
+template <typename Cursor, typename Wand>
+class BlockMaxScoredCursor: public ScoredCursor<Cursor> {
+  public:
+    using base_cursor_type = Cursor;
+
+    BlockMaxScoredCursor(
+        Cursor cursor,
+        TermScorer term_scorer,
+        float query_weight,
+        float max_score,
+        typename Wand::wand_data_enumerator wdata)
+        : ScoredCursor<Cursor>(std::move(cursor), std::move(term_scorer), query_weight, max_score),
+          m_wdata(std::move(wdata))
+    {}
+    BlockMaxScoredCursor(BlockMaxScoredCursor const&) = delete;
+    BlockMaxScoredCursor(BlockMaxScoredCursor&&) = default;
+    BlockMaxScoredCursor& operator=(BlockMaxScoredCursor const&) = delete;
+    BlockMaxScoredCursor& operator=(BlockMaxScoredCursor&&) = default;
+    ~BlockMaxScoredCursor() = default;
+
+    [[nodiscard]] auto block_max_score() -> float { return m_wdata.score(); }
+    [[nodiscard]] auto block_max_docid() -> std::uint32_t { return m_wdata.docid(); }
+    void block_max_move(std::uint32_t docid) { m_wdata.next_geq(docid); }
```

Comment:
I'd prefer this to be called `block_max_next_geq` to keep the terminology consistent between the wand data and the actual postings lists, thoughts?

Reply:
Yeah, probably you're right. I'll change that.